woocommerce/woocommerce-android
# [Scanned and Deliver] Improve permission handling flow

Closes: #9345 


### Description

This PR improves the permission handling flow in the scanning screen. 

### Main changes
1. When the user denies the permission for the first time, Notify the user with an alert dialog explaining the need for camera access for scanning the barcode.
2. Provide the user with a CTA to enable the camera permission in the above case.
3. When the user denies the permission for the second time, Notify the user with an alert dialog mentioning that they have permanently disabled the camera access for the app and they cannot scan the barcode without camera permission.
4. Provide the user with a CTA to go to the app system settings to enable the camera permission from there.
5. When the user navigates back from the app system settings, re-check the camera permission and if eligible, ask for camera permission again.

### Testing instructions


1. Open WooCommerce app
2. Navigate to the `orders` tab
3. Click on the barcode icon in the toolbar
4. **Ensure it asks for the camera permission**
5. Clicking on deny **should display an alert dialog asking to grant camera permission** with a CTA `grant`
6. Clicking on the CTA **should ask the camera permission again**
7. Denying again results in **displaying another alert dialog that says that you have permanently disabled camera permission** and to enable it by going to app system settings. Provides CTA `Go to settings`
8. Clicking on the CTA **should open WooCommerce app settings**
9. Changing the permission from the settings and **coming back to the app should ask for permission again** (if eligible)
10. Allowing the camera permission should start displaying the camera preview

### Images/gif


https://github.com/woocommerce/woocommerce-android/assets/1331230/7bd2f8dc-1010-440c-a1e7-8d68ac42cfd4
